### DISCUSSION TITLE
Hide and show menu items

### DISCUSSION THREAD

**Discussion by nickloris** [davetcc/tcMenu#499] [Q&A] [answered]
Hi,
For my project I need to hide a menu item when pressed and show another one insead.

I used the following commands to do that:

menuItem1.setVisible(false);
menuItem1.setVisible(true);

By the way the screen seems not updated to the new layout.
Hided and unided items are updated only when I exit from the submenu end re-enter.

Is that a way or a command that updates the display?

Thanks
Nicola

**Comment by davetcc** [comment, marked as answer]
Hi sorry for the delay in answering, we've been a bit busy in a rather large migration project (see other new discussion).

Whenever you do either make an item visible or hidden, or you otherwise materially change the layout of items you need to reset the menu as follows:

1. tell everyone the menu structure has changed

    menuMgr.notifyStructureChanged();

2. you may need to make the menu completely recalculate as well.

    menuMgr.resetMenu(true);

Thanks,
Dave.
